pkgs/servers/search/weaviate/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "weaviate";
+  version = "1.17.0";
+
+  src = fetchFromGitHub {
+    owner = "semi-technologies";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-cHGJ0nxV4tQwsoDDq08uFj8AXc0q7Vvy5l7eVG/Qo2A=";
+  };
+
+  vendorSha256 = "sha256-WkuD9DQFTslTbjjSaS9iLPm79Aay8g7pP/KEBWwo/Z4=";
+
+  subPackages = [ "cmd/weaviate-server" ];
+
+  postInstall = ''
+    mv $out/bin/weaviate-server $out/bin/weaviate
+  '';
+
+  meta = with lib; {
+    description = "The ML-first vector search engine";
+    longDescription = ''
+      Due to upstream making configuration via --config-file argument deprecated, configuration via
```

Comment:
This is a long description, not a tutorial. (And even for a tutorial this is far from being useful.)

Worse, this is the first release of this into Nixpkgs. There is no such a thing as "past" from our perspective. Deprecation? Config file option? What is this?

Remove this entirely.

Reply:
I agree it is not a longDescription. However, just running the binary without flags results in errors like 

```json
{"action":"config_load","config_file_path":"./weaviate.conf.json","level":"info","msg":"no config file specified, using default or environment based","time":"2022-12-24T23:00:00"}
```

Which suggests to the user about the `./weaviate.conf.json` file (also running `weaviate --help` does the same.)

However, when you try that you get a deprecation warning -

```json
{"deprecation":{"apiType":"Configuration","id":"config-files","locations":["--config-file=\"\""],"mitigation":"Configure Weaviate using environment variables.","msg":"use of deprecated command line argument --config-file","sinceTime":"2020-09-08T09:46:00.000Z","sinceVersion":"0.22.16","status":"deprecated"},"level":"warning","msg":"use of deprecated command line argument --config-file","time":"2022-12-24T23:03:39"}
```

To make it more user-friendly, this information should exist somewhere. In general, I think nix packages should have an optional `Notes` meta attribute. I agree longDescription does not fit that.

I will remove this anyway.

Reply:
As far as I remember, some quirks about specific softwares are documented in the Nixpkgs manual itself. Things like LAPACK, URxvt, Emacs, the Linux kernel, X.Org etc.

https://nixos.org/manual/nixpkgs/stable/#chap-packages